botocore/credentials.py

```diff
@@ -550,19 +554,51 @@ def load(self):
         """
         if self._mapping['access_key'] in self.environ:
             logger.info('Found credentials in environment variables.')
-            access_key, secret_key = self._extract_creds_from_mapping(
-                self.environ, self._mapping['access_key'],
-                self._mapping['secret_key'])
-            token = self._get_session_token()
-            return Credentials(access_key, secret_key, token,
-                               method=self.METHOD)
+            fetcher = self._create_credentials_fetcher()
+            credentials = fetcher()
+            expiry_time = credentials['expiry_time']
+            if expiry_time is not None:
+                expiry_time = parse(expiry_time)
+            return RefreshableCredentials(
```

Comment:
I don't think we should rely on no expiry time meaning non-refreshable credentials.  I would prefer we're explicit.  If we get expiry time use RefreshableCredentials, otherwise use Credentials.

I'm guessing that code was put in place to protect against malformed responses and not necessarily as part of its contract.  It's always confusing to have RefreshableCredentials not actually refresh if you don't provide expiry.